### PR TITLE
Namespace Set: symmetric difference methods

### PR DESCRIPTION
**Pull request by Smoren**
Hi @markrogoyski,

It's a draft of PR.

I've added `Set::symmetricDifference()` and covered it by tests.

Like the intersection methods, symmetrical difference works both with sets of unique values and with arrays in which values are repeated.

Case with repeats:

```php
use IterTools\Set;

$a = [1, 2, 3, 1];
$b = [1, 2, 3, 4];

Set::symmetricDifference($a, $b); // [1, 4]
```

* `4` is contained in the difference because it was not found in the first array.
* `1` is contained in the difference, because it was encountered twice in the first array, and only once in the second.

In case of more times repetitions:

```php
use IterTools\Set;

$a = [1, 2, 3, 1, 1];
$b = [1, 2, 3, 4];

Set::symmetricDifference($a, $b); // [1, 4, 1]
$a = [1, 2, 3, 1, 1];
$b = [1, 2, 3, 1, 4];

Set::symmetricDifference($a, $b); // [1, 4]
```

If you like this functionality I will add strict method for symmetric difference and add such methods to the `Stream` namespace.

BTW, In the case of intersection methods, the logic for accounting for repetitions is similar.

Note: this PR is depends of [the previous one](https://github.com/markrogoyski/itertools-php/pull/16).

UPD: `Set::symmetricDifferenceStrict()` method added and covered by tests.

### CODE REVIEW

**Comment by Smoren** [comment]
UPD:

1. "Symmetric difference" methods renamed:
* `Set::symmetricDifference()` => `Set::symmetricDifferenceNonStrict()`;
* `Set::symmetricDifferenceStrict()` => `Set::symmetricDifference()`.
2. Tests renamed for these methods.

**Comment by Smoren** [comment]
UPD: this branch is rebased from `set_intersection`.

**Comment by Smoren** [comment]
1. Branch is rebased from `set_intersection`.
2.  `Set::symmetricDifferenceNonStrict()` is renamed to `Set::symmetricDifferenceCoercive()`.

**Comment by Smoren** [comment]
Hi @markrogoyski,

I think this PR is ready now.

Summary:

1. New methods added:
  * `Set::symmetricDifference()`;
  * `Set::symmetricDifferenceCoercive()`;
  * `Stream::symmetricDifferenceWith()`;
  * `Stream::symmetricDifferenceCoerciveWith()`.
2. All the methods are covered by tests and documented in README file.

File: tests/Set/SymmetricDifferenceTest.php [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+                    [1, 2, 3, '4', '5'],
+                ],
+                ['1', 1, '2', 2, '3', 3, 4, '4', 5, '5'],
+            ],
+            [
+                [
+                    [1, 1, 2, 2, 1, 1],
+                    [2, 2, 1, 1, 2, 2],
+                ],
+                [1, 1, 2, 2],
```
Comment (markrogoyski) [inline tests/Set/SymmetricDifferenceTest.php:87]: Question: How are you defining "difference" in this PR? If this was set symmetric difference I'd expect this to result in an empty set. Are you defining difference the difference at each index of ordered lists of values? And if that is the case, we aren't really dealing with sets then.

Comment (markrogoyski) [reply tests/Set/SymmetricDifferenceTest.php:87]: And I suppose I have the same question for intersection now that I think about it and looking again at the unit tests in this light.

Comment (Smoren) [reply tests/Set/SymmetricDifferenceTest.php:87]: Resolved in another PR (Multisets logic).

**Comment by Smoren** [comment]
@markrogoyski,

1. Simmetric difference tests: `assertEqualsCanonicalizing` have been used.
2. README: note about multiset added for all the Symmetric difference methods.

**Pull request merged**